config_builder.py

```diff
@@ -0,0 +1,251 @@
+        _target = '{}/{}.yaml'.format(self.conf_d, integration)
+        _data = {}
+        # Do not overwrite target if exists
+        if isfile(_target):
+            _source = _target
+        with open(_source, 'r') as _stream:
+            _data = yaml.load(_stream)
+
+        # Hack to not end up with 'null' as init_config value
+        if _data.get('init_config') is None:
+            _data['init_config'] = {}
+
+        _instances = _data.get('instances')
+        for _inst in _instances:
+            _inst[property_key] = property_value
+
+        with open(_target, 'w') as _stream:
+            yaml.dump(_data, _stream, default_flow_style=False)
+
+    def build_integration_conf(self):
+        '''
+        Builds the integration configuration files from their corresponding
+        example files
+        '''
+        if getenv('KUBERNETES') is not None:
+            # allows to disable kube_service tagging if needed (big clusters)
+            _ktags = getenv('KUBERNETES_COLLECT_SERVICE_TAGS')
+            if _ktags is not None:
+                self.set_instance_property('kubernetes', 'collect_service_tags', _ktags)
+            # enable event collector
+            # WARNING: to avoid duplicates, only one agent at a time across the entire
+            # cluster should have this feature enabled.
+            if getenv('KUBERNETES_COLLECT_EVENTS') is not None:
+                self.set_instance_property('kubernetes', 'collect_events', 'true')
+                _kregex = getenv('KUBERNETES_NAMESPACE_NAME_REGEX')
+                if _kregex is not None:
+                    self.set_instance_property('kubernetes', 'namespace_name_regexp', _kregex)
+
+        if getenv('MESOS_MASTER') is not None:
+            self.set_instance_property('mesos_master', 'url', 'http://leader.mesos:5050')
+            self.set_instance_property('zk', 'host', 'leader.mesos')
+
+        if getenv('MESOS_SLAVE') is not None:
+            self.set_instance_property('mesos_slave', 'url', 'http://{}:5051'.format(getenv('HOST', 'localhost')))
+
+        if getenv('MARATHON_URL') is not None:
+            _murl = getenv('MARATHON_URL')
+            if _murl is not None:
+                self.set_instance_property('marathon', 'url', _murl)
+
+    def load_config(self, config_file):
+        '''
+        Loads a config file using ConfigParser
+        '''
+        self.config = ConfigParser()
+        # import existing config from file
+        with open(config_file, 'rb') as cfd:
+            self.config.readfp(cfd)
+
+    def save_config(self, config_file):
+        '''
+        Saves a ConfigParser object (self.config) to the given file
+        '''
+        if self.config is None:
+            logging.error('config object needs to be created before saving anything')
+            exit(1)
+        with open(config_file, 'wb') as cfd:
+            self.config.write(cfd)
+
+    def build_supervisor_conf(self):
```

Comment:
3 sed in 6 dockerfile and 6 entrypoint.sh (for now)....
Why doing things 3 different ways? If we do a python script for maintaining part of the config, why not reuse the function for the rest?
Right now supervisor.conf has 3 parameters changed, meaning 3 sed.
How maintainable will it be when you want to change the 10 others?